src/java/io/bazel/rulesscala/test_discovery/FilteredRunnerBuilder.scala

```diff
@@ -0,0 +1,33 @@
+package io.bazel.rulesscala.test_discovery
+
+import java.util.regex.Pattern
+import org.junit.runner.Runner
+import org.junit.runners.BlockJUnit4ClassRunner
+import org.junit.runners.model.FrameworkMethod
+import org.junit.runners.model.RunnerBuilder
+import scala.collection.JavaConverters._
+
+object FilteredRunnerBuilder extends RunnerBuilder {
+  // Defined by --test_filter bazel flag.
+  private val pattern = sys.env.get("TESTBRIDGE_TEST_ONLY").map(Pattern.compile(_))
+
+  override def runnerForClass(testClass: Class[_]): Runner =
+    pattern match {
+      case Some(pattern) => new FilteredClassRunner(testClass, pattern)
```

Comment:
btw, you can live it as is but just as an FYI binding in the case to the same name of the matched variable is a bit of a scala anti practice since you now have shadowing.
What many people do (and is not amazing but works) is to call you `private val pattern...` `private val maybePattern`